src/pip/_internal/cli/base_command.py

```diff
@@ -257,6 +260,25 @@ def main(self, args):
 
 class RequirementCommand(Command):
 
+    @staticmethod
+    def make_requirement_preparer(directory,   # type: TempDirectory
+                                  options,     # type: Any
```

Comment:
`options` should be type `Values`.

You're also going to want to add optional keyword arguments `download_dir=None` and `wheel_download_dir=None` which each of the other commands will use.